paparazzi/api/paparazzi.api

```diff
@@ -0,0 +1,255 @@
+	public fun equals (Ljava/lang/Object;)Z
+	public final fun getDensity ()Lcom/android/resources/Density;
+	public final fun getFolderConfiguration ()Lcom/android/ide/common/resources/configuration/FolderConfiguration;
+	public final fun getFontScale ()F
+	public final fun getHardwareConfig ()Lcom/android/ide/common/rendering/api/HardwareConfig;
+	public final fun getKeyboard ()Lcom/android/resources/Keyboard;
+	public final fun getKeyboardState ()Lcom/android/resources/KeyboardState;
+	public final fun getLayoutDirection ()Lcom/android/resources/LayoutDirection;
+	public final fun getLocale ()Ljava/lang/String;
+	public final fun getNavigation ()Lcom/android/resources/Navigation;
+	public final fun getNightMode ()Lcom/android/resources/NightMode;
+	public final fun getOrientation ()Lcom/android/resources/ScreenOrientation;
+	public final fun getRatio ()Lcom/android/resources/ScreenRatio;
+	public final fun getReleased ()Ljava/lang/String;
+	public final fun getScreenHeight ()I
+	public final fun getScreenRound ()Lcom/android/resources/ScreenRound;
+	public final fun getScreenWidth ()I
+	public final fun getSize ()Lcom/android/resources/ScreenSize;
+	public final fun getSoftButtons ()Z
+	public final fun getTouchScreen ()Lcom/android/resources/TouchScreen;
+	public final fun getUiMode ()Lcom/android/resources/UiMode;
+	public final fun getUiModeMask ()I
+	public final fun getXdpi ()I
+	public final fun getYdpi ()I
+	public fun hashCode ()I
+	public fun toString ()Ljava/lang/String;
+}
+
+public final class app/cash/paparazzi/DeviceConfig$Companion {
+	public final fun getEnumMap (Ljava/io/File;)Ljava/util/Map;
+	public final fun loadProperties (Ljava/io/File;)Ljava/util/Map;
+}
+
+public final class app/cash/paparazzi/Environment {
+	public static final field $stable I
+	public fun <init> (Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;ILjava/util/List;Ljava/util/List;Ljava/util/List;Ljava/util/List;Ljava/util/List;Ljava/util/List;)V
+	public final fun component1 ()Ljava/lang/String;
+	public final fun component10 ()Ljava/util/List;
+	public final fun component2 ()Ljava/lang/String;
+	public final fun component3 ()Ljava/lang/String;
+	public final fun component4 ()I
+	public final fun component5 ()Ljava/util/List;
+	public final fun component6 ()Ljava/util/List;
+	public final fun component7 ()Ljava/util/List;
+	public final fun component8 ()Ljava/util/List;
+	public final fun component9 ()Ljava/util/List;
```

Comment:
Do you really want to publish this? adding a prop will be breaking unless it's last.

Reply:
(applies to other classes too)

Reply:
[copy is useful though](https://github.com/TWiStErRob/net.twisterrob.sun/blob/8bd175671ae513cb090e1b65284322c2384ca80e/component/paparazzi/src/main/java/net/twisterrob/sun/test/screenshots/PaparazziFactory.kt#L43-L46)... tricky one.

Reply:
yea, I think that's fine?  since this would also be a constructor argument change, which unless we have default values, would also be a breaking change.  what do other libraries typically do here w.r.t. exposing data classes?

Reply:
I think they don't, but @JakeWharton probably have a lot of experience with this.

Reply:
Addressed overall in this comment (below): https://github.com/cashapp/paparazzi/pull/1264#discussion_r1469102437

Reply:
data classes should never be used in public library API